From: Nathaniel <me@nathanielbennett.com>
Date: Tue, 8 Jun 2021 20:02:52 -0400
Subject: Fixed 'reverse' macro modifier
Origin: https://github.com/shevek/libspf2/commit/f06fef6cede4c4cb42f2c617496e6041782d7070

---
 src/libspf2/spf_expand.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/libspf2/spf_expand.c b/src/libspf2/spf_expand.c
index 25689da11e84..f93005d26f44 100644
--- a/src/libspf2/spf_expand.c
+++ b/src/libspf2/spf_expand.c
@@ -86,7 +86,7 @@ SPF_record_expand_data(SPF_server_t *spf_server,
 {
 	SPF_data_t	*d, *data_end;
 
-	size_t		 len;
+	size_t		 len, label_len;
 	const char	*p_err;	// XXX Check this value, when returned.
 	char		*p, *p_end;
 	const char	*p_read;
@@ -300,9 +300,9 @@ top:
 				if ( SPF_delim_valid(d, *p_read) ) {
 					/* Subtract 1 because p_read points to delim, and
 					 * p_read_end points to the following delim. */
-					len = p_read_end - p_read - 1;
-					memcpy( p_write, p_read + 1, len );
-					p_write += len;
+					label_len = p_read_end - p_read - 1;
+					memcpy( p_write, p_read + 1, label_len );
+					p_write += label_len;
 					*p_write++ = '.';
 
 					p_read_end = p_read;
@@ -314,9 +314,9 @@ top:
 			 * string. p_read_end might also point there if the string
 			 * starts with a delimiter. */
 			if (p_read_end >= p_read) {
-				len = p_read_end - p_read - 1;
-				memcpy( p_write, p_read + 1, len );
-				p_write += len;
+				label_len = p_read_end - p_read - 1;
+				memcpy( p_write, p_read + 1, label_len );
+				p_write += label_len;
 				*p_write++ = '.';
 			}
 
-- 
2.20.1

